### DISCUSSION TITLE
Semaphore's development prospect

### DISCUSSION THREAD

**Discussion by fiftin** [semaphoreui/semaphore#1111] [General]
Semaphore's community is growing, the number of wishes too.

We need backers to continue the development of the project at a fairly high pace. This was not required before, since the Semaphore was in my working stack. Now this is not so (like everyone else, I employee and can't always choose a stack). Now I have a difficult choice - to spend my free time on freelance or lose money, but spend it on a useful project.

You can becomes backers on https://ko-fi.com/fiftin. Or may be you have another idea? :)

**Comment by cf-sewe** [comment]
Thats unfortunate, but understandable 👍 
Are you the only responsible for this project? I am concerned that if insufficient funds come in, the project will silently die due to lack of support.  

If you are the only main responsible for this project, have you considered getting more contributors in the boat?
Alternatively, Semaphore could also be brought to a commercial level (fork + license change), but that would require a different level of contribution of course.

**Reply by fiftin** [reply]
> the project will silently die due to lack of support

I will do my best to prevent this from happening.

**Reply by fiftin** [reply]
> Alternatively, Semaphore could also be brought to a commercial level 

This is one of the options I'm considering, but first I want to try to do without it.

**Reply by fiftin** [reply]
> If you are the only main responsible for this project, have you considered getting more contributors in the boat?

This is not reliable, the project has already passed this in 2018. Coordinating the work of a team of poorly motivated developers takes too much time and effort.

**Comment by ansibleguy** [comment]
Greetings!

I would like to 'resurrect' this discussion.

I've invested (_just_) around 20-30h of support/documentation/fixing to this project up until now.

In my opinion this project **could be great** but cannot be used by many users as it:
* lacks major functionality
* has several breaking bugs
* could have better documentation
* could benefit from [community channels](https://github.com/ansible-semaphore/semaphore/discussions/1410)

As you @fiftin do seem to have very little time for this project (_no judgement_) I would like to ask where you see this project in the future?
* This workload may only be manageable in a team/community.
* Will you allow other people to co-maintain the repository and help you out?
* Would you like to see a community-driven fork?

The pace at which it is developed at the moment is a little sad to see.
Also: it is very frustrating that contributions (_especially PR's_) are not reviewed at all (_after months_). Even a 'deny-comment' would be better than no response.

To finish on something positive:
* As this project is so lightweight it can enable so many non-CLI users to use Ansible and therefore can have a great user-base
* There seem to be some people that are passionate about this project

Thank you for your work.

\- AnsibleGuy

**Reply by ahoiroman** [reply]
For me, the current state is a big risk, as a project maintained by one person has a big risk to be abandoned some time (no offense, I know open source does not provide food and rent in most cases).

As I am not a Go programmer, I cannot provide my skills - but I can help with some sponsorship (even if it's not much) and infrastructure/hosting.

**Reply by ansibleguy** [reply]
@ahoiroman I agree with your standpoint!
As you mentioned 'big risk' - there is also the risk of security-fixes not being shipped in a timely manner.

**Reply by ansibleguy** [reply]
I'm sorry, but @fiftin could you give us some of your thought regarding this topic?
I really would like to invest some hours every month for contributing to this project - but if nothing changes, it could be a waste of my (precious) resources..

**Reply by fiftin** [reply]
Hi @ansibleguy I understand you. Thank you very much for your efforts. I didn't have time to work on the project at all (for good reasons), but now I'm working on it all my free time.  So, if you have time you are welcome.

**Reply by fiftin** [reply]
@ahoiroman any sponsorship matters.

**Reply by fiftin** [reply]
> For me, the current state is a big risk, as a project maintained by one person has a big risk to be abandoned some time 

any project has this risk ;)

**Reply by fiftin** [reply]
@ansibleguy And I think now you understand that maintaining a project is more difficult than it seems ;) It always require more time than you have.
